File: src/timing.py
#!/usr/bin/env python
import time
import threading

print "Pyton sleep and timestamp tests:"

for i in xrange(0, 10):
    start = time.time()
    time.sleep(0.000000001)
    end = time.time()
    duration = long((end - start) * 1000000000.0)

    print "Sleep Duration: " + str(duration) + " ns, expeced 1 ns"

for i in xrange(0, 10):
    start = time.time()
    end = time.time()
    duration = long((end - start) * 1000000000.0)

    print "Timestamp took " + str(duration) + " ns, expeced 1 ns"

File: src/Timing.java
import util.PrintUtil;

public class Timing
{
	public static void main(String[] args) throws Exception
	{
		System.out.println("Java sleep and timestamp tests:");
		
		for (int i = 0; i < 10; i++)
		{
			long start = System.nanoTime();
			Thread.sleep(0, 1);
			long end = System.nanoTime();
			long duration = end - start;
			
			PrintUtil.print("Sleep Duration: " + duration + " ns, expeced 1 ns");
		}
		
		for (int i = 0; i < 10; i++)
		{
			long start = System.nanoTime();
			long end = System.nanoTime();
			PrintUtil.print("Timestamp took " + (end - start) + " ns");
		}
	}
}
